orderBook.js

```diff
@@ -0,0 +1,49 @@
+function reconcileOrder(existingBook, incomingOrder) {
+  // if the existing book does not contain any data
+  if (existingBook.length === 0) {
+    // add incoming order to empty book
+    existingBook.push(incomingOrder)
+
+    return existingBook
+  }
+  let newArray = []
+
+  // run through the items in the existing book
+  for (let i = 0; i < existingBook.length; i++) {
+    // turn current item in existingBook into a variable so it can be reused
+    let currentItem = existingBook[i]
+
+    // if the order type is the same add it to the existing book
+    // if the order type is different also add it
+    if (incomingOrder.type !== currentItem.type &&
+      (incomingOrder.price === currentItem.price ||
+      (incomingOrder.type === 'sell' && incomingOrder.price < currentItem.price))) {
```

Comment:
can probably extract if check to a seperate function

incomingOrder.type === 'sell' && incomingOrder.price < currentItem.price) can be put in its own function and return true or false.